app/scripts/controllers/metametrics.js

```diff
@@ -605,6 +606,27 @@ export default class MetaMetricsController {
     return unique.length;
   }
 
+  /**
+   * @param {object} metamaskState
+   * @returns number of unique token addresses
+   */
+  _getNumberOfTokens(metamaskState) {
+    const allUniqueAddresses = Object.values(metamaskState.allTokens).reduce(
+      (result, tokensByAccountByChain) => {
+        Object.values(tokensByAccountByChain).forEach((tokensByAccount) => {
+          tokensByAccount.forEach((token) => {
+            result.add(token.address);
+          });
+        });
+
+        return result;
+      },
+      new Set(),
+    );
+
+    return allUniqueAddresses.size;
+  }
```

Comment:
I think I got this right... "tokensByAccountByChain" is what I presumed from looking through the TokensController and the data produced. Please double-check me here 🙏 

I am also wondering two other things:
1. How large could the data set get, and would it be worth it to use lodash memoize? 
2. May we want to move this logic into the TokensController someday?

Reply:
1. memoizing functions might be worth while, but that can happen down the road. We should discuss the proper level of memoization (should we memoize the entire build user trait function, only portions of the traits, etc)
2. No, although it seems like we will end up with a bunch of methods related to other parts of state, they are building user traits which is specific to meta metrics. I feel all user traits should be built and established inside the metametrics controller regardless of where in state they come from

Reply:
Thank you for the thoughts! 

1. 👍 
2. I thought, maybe, the total number of tokens could be shown in other places. Thanks for confirming that this method wouldn't be desirable in the TokenController. I am all for keeping the user trait logic inside the MetaMetrics controller